src/reservoir/Reservoir.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  createClient,
+  getClient,
+  // @ts-ignore
+  // eslint-disable-next-line import/no-unresolved
+} from "@reservoir0x/reservoir-sdk";
+import {
+  Address,
+  createPublicClient,
+  getContract,
+  Hash,
+  http,
+  PublicClient,
+} from "viem";
+import { mainnet } from "viem/chains";
+
+import { Wallet, zeroHash } from "@/blockchain";
+import { erc721ABI } from "@/generated/blockchain/v5";
+
+import {
+  adaptWalletToCaptureTxData,
+  ETH_CONTRACT_ADDRESS,
+  generateExpectedCurrencyPriceObject,
+  generateMatchOrdersExecutionData,
+  getExecutionData,
+  SeaportAskOrBid,
+  WETH_CONTRACT_ADDRESS,
+} from "./utils";
+
+export class Reservoir {
+  baseApiUrl: string;
+  mainnetClient: PublicClient;
+  wallet: Wallet;
+
+  constructor({
+    baseApiUrl = "https://api.reservoir.tools",
+    apiKey,
+    infuraApiKey,
+    wallet,
+  }: {
+    baseApiUrl?: string;
+    apiKey?: string;
+    infuraApiKey?: string;
+    wallet: Wallet;
+  }) {
+    this.baseApiUrl = baseApiUrl;
+    this.wallet = wallet;
+
+    createClient({
+      chains: [
+        {
+          id: wallet.chain.id,
+          name: wallet.chain.name,
+          baseApiUrl,
+          active: true,
+        },
+      ],
+      apiKey,
+      source: "gondi.xyz",
+    });
+
+    this.mainnetClient = createPublicClient({
+      chain: mainnet,
+      transport: http(`https://mainnet.infura.io/v3/${infuraApiKey}`),
+    });
+  }
+
+  async getAsk({ orderId }: { orderId: Hash }) {
+    return fetch(
```

Comment:
We could type using https://docs.reservoir.tools/reference/typescript-api-typings#parsing-api-responses